### ISSUE TITLE
[HOLD for payment 2023-11-21] [$500]  Removed users are not being displayed on the invite page

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#30274]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:**
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:**

## Action Performed:
1. Click 'FAB' > Start Chat > create a #Room
2. Complete the Room details > click Create Room
4. Click on the Room details in the new chat > Members > click Invite and invite several test accounts
5. Click Invite
6. Now Select several members and click Remove (remember the email of the users you remove) > accept the popup
7. Click the Invite button again
8. Notice that removed members are not listed as members to invite back to the workspace

## Expected Result: 
After being Removed from a workspace, you should be able to invite the same users back to the workspace but they are not listed after clicking Invite

## Actual Result: 
Removed members should be showing after you click on the Invite button on the "find a member" screen

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/38435837/a737810c-27ac-42a9-aa9f-70bada2c2a14

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/38435837/716de160-4d23-416b-97ae-fad2739db35c

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/38435837/15820999-236b-4ee7-84b6-8583e76fcf6a

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/38435837/6c40ebb6-7594-40fb-9a87-80d173b907cd

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/38435837/5f3a487a-edcb-4f25-b6c6-a832f20e794a

https://github.com/Expensify/App/assets/38435837/5d68773a-4001-4130-a8ab-2934e04912a5

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/38435837/8ced8b25-11b2-4aff-89a2-666b7c1bbfce

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~017c6417380b857a15</li>
        <li>Upwork Job ID: 1716852669249048576</li>
        <li>Last Price Increase: 2023-10-31</li>
<li>Automatic offers: </li>
<ul>
<li>tienifr | Contributor | 27501792</li>
</ul></li>
    </ul>
</details>

**Comment by c3024** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
Removed members from room page are not shown in invite page to invite and searching for them also yields no results
### What is the root cause of that problem?
When we create a room, the report contains only `participantAccountIDs`.
But when we visit a members page of a room, BE sends `participants` (which are logins) as well and merge them with report. Going to invite page does not fetch any new data from BE.
This logic here 
https://github.com/Expensify/App/blob/d28247332f58bb0d4296b0eab49527771c03464c/src/pages/RoomInvitePage.js#L72
https://github.com/Expensify/App/blob/d28247332f58bb0d4296b0eab49527771c03464c/src/pages/RoomInvitePage.js#L85
needs logins so the participants fetched when visiting Members page is required.
However, when we remove a user, the `RemoveFromRoom` does not return anything from BE. 
This here 
https://github.com/Expensify/App/blob/d28247332f58bb0d4296b0eab49527771c03464c/src/libs/actions/Report.js#L2027-L2066
updates only `participantAccountIDs` and not `participants` in Onyx so the excluded user logins include the remove member login.
So invite options do not show this user because they are in the excluded users.
### What changes do you think we should make in order to solve the problem?
We should get `participants` as well here
https://github.com/Expensify/App/blob/d28247332f58bb0d4296b0eab49527771c03464c/src/libs/actions/Report.js#L2024
by changing it to
```js
    const {participantAccountIDs, participants} = report;
```
and add participants after removal as well here
https://github.com/Expensify/App/blob/d28247332f58bb0d4296b0eab49527771c03464c/src/libs/actions/Report.js#L2026
by adding this
```js
const participantsAfterRemoval = _.map(participantAccountIDsAfterRemoval, (accountID) => allPersonalDetails[accountID].login);
```
or get with the `_.difference` logic similar to accountIDs after removal.
and change this
https://github.com/Expensify/App/blob/d28247332f58bb0d4296b0eab49527771c03464c/src/libs/actions/Report.js#L2027
to
```js
    API.write(
        'RemoveFromRoom',
        {
            reportID,
            targetAccountIDs,
        },
        {
            optimisticData: [
                {
                    onyxMethod: Onyx.METHOD.MERGE,
                    key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
                    value: {
                        participantAccountIDs: participantAccountIDsAfterRemoval,
                        participants: participantsAfterRemoval
                    },
                },
            ],
            failureData: [
                {
                    onyxMethod: Onyx.METHOD.MERGE,
                    key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
                    value: {
                        participantAccountIDs,
                        participants
                    },
                },
            ],

            // We need to add success data here since in high latency situations,
            // the OpenRoomMembersPage call has the chance of overwriting the optimistic data we set above.
            successData: [
                {
                    onyxMethod: Onyx.METHOD.MERGE,
                    key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
                    value: {
                        participantAccountIDs: participantAccountIDsAfterRemoval,
                        participants: participantsAfterRemoval
                    },
                },
            ],
        },
    );
}
```
### What alternative solutions did you explore? (Optional)
<details>
<summary>Result</summary>


https://github.com/Expensify/App/assets/102477862/0f39aa18-d0ee-45d0-8b8c-a6ed6e4ab4a5







</details>

**Comment by Christinadobrzyn** [comment]
similar to this, we're intentionally not including 'removal' details in the audit logs. https://github.com/Expensify/App/issues/29979

So this is a feature request that is on hold

**Comment by Christinadobrzyn** [comment]
Ah okay, but I don't think this is a bug. Reason:

1. When you create a room > click the room settings > click Members you only see the Admin. 

![image](https://github.com/Expensify/App/assets/51066321/ef984d55-20fa-4269-8274-0b1dff720f3d)

2. When you remove members from the room, you only see the members who are still part of the room. 

![image](https://github.com/Expensify/App/assets/51066321/906bbfc7-ef28-4c14-8f2c-c2de84dea1eb)

It's not until you click "Invite" do you see more members to invite to the room. This is consistent with both Inviting and Removing members. 

@tienifr will you let me know if you see otherwise?

**Comment by Christinadobrzyn** [comment]
Ah that is helpful @tienifr - I can reproduce that and I do think it needs to be fixed. To confirm @tienifr were you the reporter? I don't see who the reporter was in the OP.

**Comment by Christinadobrzyn** [comment]
Looks like we have a few proposals that can be reviewed when you have a moment @allroundexperts!

**Comment by Julesssss** [comment]
Agree with @allroundexperts

**Comment by Christinadobrzyn** [comment]
I'm going to be ooo on payment day so reassigning. Here's what I understand about payment on Nov 21st

@allroundexperts can you let us know about a regression test?

Payouts due on Nov 21st:

Issue Reporter: NA
Contributor: $500 @tienifr (through Upwork)
Contributor+: $500 @allroundexperts (through NewExpensify)

Eligible for 50% #urgency bonus? NA

Upwork job is [here](https://www.upwork.com/jobs/~017c6417380b857a15) (the job is closed).

**Comment by joekaufmanexpensify** [comment]
[Bumped in slack ](https://expensify.slack.com/archives/C02NK2DQWUX/p1700490543589399)

**Comment by joekaufmanexpensify** [comment]
Great, ty! BZ checklist is all set.

**Comment by joekaufmanexpensify** [comment]
All set to issue payment. @allroundexperts could you please request $500 via NewDot and comment here once complete?

**Comment by joekaufmanexpensify** [comment]
@tienifr $500 sent and contract ended!

**Comment by JmillsExpensify** [comment]
$500 payment approved for @allroundexperts based on [this comment](https://github.com/Expensify/App/issues/30274#issuecomment-1817127366).
